crud/crud.cpp

```diff
@@ -1068,3 +1083,37 @@ crud::flush_expiration_entries(const bzn::uuid_t& uuid)
         }
     }
 }
+
+
+bool
+crud::operation_exceeds_available_space(const database_msg& request, const Json::Value& perms)
+{
+    const auto max_size = boost::lexical_cast<uint64_t>(perms[MAX_SIZE_KEY]);
+
+    // any max size set?
+    if (max_size)
+    {
+        uint64_t size;
+        std::tie(std::ignore, size) = this->storage->get_size(request.header().db_uuid());
+
+        // test create
+        if (request.msg_case() == database_msg::kCreate)
+        {
+            return (size + request.create().key().size() + request.create().value().size() > max_size);
+        }
+
+        // test update
+        if (request.msg_case() == database_msg::kUpdate)
+        {
+            const auto prev_kv_size = this->storage->get_key_size(request.header().db_uuid(),
+                request.update().key());
+
+            if (prev_kv_size)
+            {
+                return (size - *prev_kv_size + request.update().key().size() + request.update().value().size() > max_size);
+            }
```

Comment:
I sort of want there to be an else in this case, with the same return value as a create would give... But I realize we would never get here if there was no previous key anyway...